docs/DeveloperGuide.md

```diff
@@ -1,34 +1,346 @@
 # Developer Guide
 
+## Introduction
+
+RevisED is a command line application to help students revise by creating flashcards and taking quizzes, 
+and keeping track of their deadlines, tasks and any other important dates. This application allows users to add subjects, 
+their different topics and create different flashcards in form of question and answers on each topic. Students can also 
+take quizzes which asks them the questions that they added in form of flashcards and prompted to answer them. 
+The student can view results of these quizzes immediately and are stored so that students can view them later. 
+Additionally, students can delete, and list subjects, topics, and flashcards and find and enter subjects and topics.   
+
+This developer guide is for developers who wish to understand and/or develop ReviseED further. 
+This guide includes design, implementation, product scope, user stories, non-functional requirements, glossary, 
+testing and development sections to help developers better understand the application.
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+The application has been broken down into different segments so that users would find it easy to navigate from one
+segment to the next one. There are three segments within the application:
+* Subjects: This segment edits the list of all the subjects present within the application and takes care of relevant
+commands such as add subject, delete subject or find subject.
+* Subject: This segment is accessed through the `subject NAMEOFSUBJECT` command. The user would then look at the individual 
+subjects and then would be able to add either tasks or topics.
+* Topic : This segment is accessed through the `topic NAMEOFTOPIC` command. The user can add, delete or find flashcards
+in this segment.
+
+The code is broken down into different packages, which combine classes that perform a similar functionality. Here
+are the list of packages:
```

Comment:
I think it would be good to add a class diagram to show the structure of the classes.